src/core/Akka.Discovery/Discovery.cs

```diff
@@ -71,27 +71,65 @@ private ServiceDiscovery CreateServiceDiscovery(string method)
-            {
-                var dynamic = DynamicAccess.CreateInstanceFor<ServiceDiscovery>(typeName, _system);
-                return dynamic.RecoverWith(ex => ex is TypeLoadException or MissingMethodException 
-                    ? DynamicAccess.CreateInstanceFor<ServiceDiscovery>(typeName) 
-                    : dynamic);
-            }
-
             var className = config.GetString("class");
             _log.Info($"Starting Discovery service using [{method}] method, class: [{className}]");
-            var instanceTry = Create(className);
 
-            return instanceTry.IsSuccess switch
+            try
+            {
+                return Create(className);
+            }
+            catch (Exception ex)
+            {
+                if (ex is TypeLoadException or MissingMethodException)
+                    throw new ArgumentException(
+                        message: $"Illegal akka.discovery.{method}.class value or incompatible class!\n" +
+                                 "The implementation class MUST extend Akka.Discovery.ServiceDiscovery with:\n" +
+                                 "  * parameterless constructor, " +
+                                 $"  * constructor with a single {nameof(ExtendedActorSystem)} parameter, or\n" +
+                                 $"  * constructor with {nameof(ExtendedActorSystem)} and {nameof(Configuration.Config)} parameters.",
+                        paramName: nameof(method), 
+                        innerException: ex);
+                throw;
+            }
+
+            ServiceDiscovery Create(string typeName)
             {
-                true => instanceTry.Get(),
-                false when instanceTry.Failure.Value is TypeLoadException or MissingMethodException =>
-                    throw new ArgumentException(nameof(method), $"Illegal akka.discovery.{method}.class value or incompatible class! \n" +
-                        "The implementation class MUST extend Akka.Discovery.ServiceDiscovery and take an \n" +
-                        "ExtendedActorSystem as constructor argument."),
-                _ => throw instanceTry.Failure.Value
-            };
+                var type = Type.GetType(typeName: typeName);
+                if (type is null || !typeof(ServiceDiscovery).IsAssignableFrom(type))
+                    throw new TypeLoadException();
+
+                var bindFlags = BindingFlags.Instance | BindingFlags.Public;
+                
+                var ctor = type.GetConstructor(
+                    bindingAttr: bindFlags,
+                    binder: null,
+                    types: new[]
+                    {
+                        typeof(ExtendedActorSystem), 
+                        typeof(Configuration.Config)
+                    }, 
+                    modifiers: null);
+                if (ctor is not null)
+                    return (ServiceDiscovery) Activator.CreateInstance(type, _system, config);
```

Comment:
New way to instantiate `ServiceDiscovery` that pushes the `Config` into the constructor